zebra-rpc/src/methods.rs

```diff
@@ -1069,59 +1019,38 @@ where
                     confirmations,
                     verbose,
                 )),
-                zebra_state::ReadResponse::Transaction(None) => Err(Error {
-                    code: ErrorCode::ServerError(0),
-                    message: "Transaction not found".to_string(),
-                    data: None,
-                }),
+                zebra_state::ReadResponse::Transaction(None) => {
+                    Err(to_error("Transaction not found"))
+                }
                 _ => unreachable!("unmatched response to a transaction request"),
             }
         }
         .boxed()
     }
 
-    // TODO:
-    // - use a generic error constructor (#5548)
-    // - use `height_from_signed_int()` to handle negative heights
-    //   (this might be better in the state request, because it needs the state height)
-    // - create a function that handles block hashes or heights, and use it in `get_block()`
     fn z_get_treestate(&self, hash_or_height: String) -> BoxFuture<Result<GetTreestate>> {
         let mut state = self.state.clone();
+        let network = self.network.clone();
 
         async move {
             // Convert the [`hash_or_height`] string into an actual hash or height.
-            let hash_or_height = hash_or_height
-                .parse()
-                .map_err(|error: SerializationError| Error {
-                    code: ErrorCode::ServerError(0),
-                    message: error.to_string(),
-                    data: None,
-                })?;
+            let hash_or_height = hash_or_height.parse().map_err(to_error)?;
```

Comment:
Optional: If we want to get resolve the "use `height_from_signed_int()` to handle negative heights" TODO, could we replace this with something like:

```suggestion
            let hash_or_height = match hash_or_height.parse() {
                Ok(hash_or_height) => hash_or_height,
                Err(err) => height_from_signed_int(
                    hash_or_height.parse().map_err(|_| err).map_server_error()?,
                    latest_chain_tip
                        .best_tip_height()
                        .ok_or_server_error("No Chain tip available yet")?,
                )
                .map_server_error()?
                .into(),
            };
```

Optional: Could we do this in a function instead and use it in `get_block()` too?

Reply:
The docs for the `z_get_treestate` say:

_The zcashd doc reference says that the parameter "`height` can be
negative where -1 is the last known valid block". On the other hand,
`lightwalletd` only uses positive heights, so Zebra does not support
negative heights._

I added a TODO regarding the negative heights in [641fd71](https://github.com/ZcashFoundation/zebra/pull/8533/commits/641fd718a19f847e8a078f25ed63c2ab937a00cc). If there isn't an API consumer in need of the functionality, I'd lean towards not implementing it.

If we decide to implement it, I'd prefer to do it in a different PR to keep the scope of the current one small.